ADR/0004-component-image-location.md

```diff
@@ -0,0 +1,187 @@
+4. Allowlist user 'shbose' to be create organizations using non-user-tokens using the Quay.io API.
+
+| Syntax      | Description |
+| ----------- | ----------- |
+| Quay.io organization      | quay.io/redhat-user-workloads       |
+| OAuth Application  name | Created, name redacted        |
+| Account used to generate token | `shbose` |
+| Scope | Administer organizations, Adminster repositories, Create Repositories |
+
+<img width="1313" alt="image" src="https://user-images.githubusercontent.com/545280/212758440-5807cd4e-11b1-43bc-aa03-385b9284cb9e.png">
+
+
+#### Organization and Image Repository creation
+
+When a user creates a Component, a StoneSoup service would need to generate the image repository for consumption by the 
+build, test and deployment services.
+
+* For each user, create a new Quay.io org “quay.io/unique-org-david”
+* For each `Component` 'foo', create a new image repo “quay.io/unique-org-david/appname/componentname”
+* Configure the robot account “redhat-<resource-uuid>” in “quay.io/unique-org-david” to be able to push to “quay.io/unique-org-david/appname-foo” 
+* Configure a `Secret` in the user's namespace with the robot account token.
+* Annotate the `Component` with the name of the image repository and the name of the `Secret` containing the robot account token.
+
+The following deviations from this design would be implemented:
+
+Until the capability to progammatically create organizations in Quay.io is activated:
+* Images repositories would be created in quay.io/redhat-user-workloads
+* Isolation would continue to be maintained the same way - every image repository would have a scoped robot account that would be rotatable using Quay.io API for the same https://docs.quay.io/api/swagger/#!/robot/regenerateUserRobotToken.
+
+Until the capability to determine the associated user/tenant/Space a Component is implemented,
+* The quay.io repository created with use the format   quay.io/org-name/namespace-name/application-name/component-name. Example, https://quay.io/repository/redhat-user-workloads/shbose/applicationname/automation-repo
+
+#### Lifecycle of the Quay.io resources
+
+* Token generation: 
+    * Robot account token: At the moment, the controller responsible for generating the Quay.io resources would be responsible for rotating the tokens https://docs.quay.io/api/swagger/#!/robot/regenerateUserRobotToken
+    * Quay.io API token: No programmatic way to regenerate this token is known at this point of time. This would be a manual activity to begin with. 
+
+* Upon deletion of an `Application`/`Component` from StoneSoup, 
+    * The controller/finalizer would delete the the relevant Quay.io resources namely, the image repository and the robot account.
```

Comment:
Revisiting --

If there was a deletion delay to enable the repository to be re-hooked up to, would it even be _possible_ to create a new component which will push to the same location? Based on my current understanding, this isn't really possible without a new feature to "restore a recently deleted component." Therefore, this functionality would also be relegated to its own feature/request if desired.

> Could we support component archiving where artifacts remain but all configuration is unhooked?

Component images that have not been mirrored to some other location (i.e. via release pipelines) can be treated as "developer" images in that they are not guaranteed to remain for a long point in time. While these images can be treated a deletable (the behavior MUST be documented), an archival state seems like it could be useful as well -- but this would be a different feature/request.

> What if we have multiple components being pushed to the same repository but different versions (i.e. different branches)?

This still seems like a potentially valid use case to me. Do we afford users with enough customization and flexibility to push multiple components to one repository? I feel like we _should_ be able to afford this customization. Based on the current flow, I assume not ... so again, it would have to be a new feature request.

> I assume that this would have to vary if we are pushing to a custom quay organization.

Likely so, but this is far outside of the current support that it seems fine to defer this question until (and if) we decide to get there